src/ui/clientComponentLineDecorator/index.ts

```diff
@@ -26,31 +26,35 @@ export class ClientComponentLineDecorator {
             this.decoration = undefined;
         }
         if (node && node.type === 'client') {
+            const config = vscode.workspace.getConfiguration('nextjsComponentBoundaryVisualizer');
+            const enableLineIcon = config.get<boolean>('enableLineIcon');
             this.decoration = vscode.window.createTextEditorDecorationType({
                 isWholeLine: true,
                 backgroundColor: 'rgba(128,0,128,0.08)', // light purple
-                before: {
+                before: enableLineIcon ? {
                     contentText: typeIcon.client,
                     color: '#a259ff',
                     margin: '0 8px 0 0',
                     fontWeight: 'bold',
-                },
+                } : undefined,
```

Comment:
If you want to disable the line icon, shouldn't `new ClientComponentLineDecorator(context, graph)` in extension.ts be the one that checks for this condition?

In other words, wouldn't the `ClientComponentLineDecorator` itself not need to be executed?